test/metabase/models/setting_test.clj

```diff
@@ -622,6 +622,23 @@
              (settings-last-updated-value-in-db))))))
 
 
+;;; ---------------------------------------------- Runtime Setting Options ----------------------------------------------
+
+(def my-default 3)
+
+(defsetting test-dynamic-setting
```

Comment:
I don't think this test reflects your change since it passes without it.

Reply:
Ah, good catch! Forgot how subtle the compile time gotchas were. Kept these assertions as I think this behavior is important too, but added another assertion that shows we fixed a gotcha with settings that are used in the conditional logic.